### ISSUE TITLE
Revert proxyConfig deprecation for Next.js

### ISSUE THREAD

**Issue by luchsamapparat** [nrwl/nx#16156]
## Description
Using the `proxyConfig` option of `@nrwl/next:server` prints a warning that this option is deprecated and will be removed in the next major version:

```
Option "proxyConfig" is deprecated: Use the built-in rewrite feature from Next.js. See: https://nextjs.org/docs/api-reference/next.config.js/rewrites
The "proxyConfig" option will be removed in Nx 16. Use the "rewrites" feature from Next.js instead. See: https://nextjs.org/docs/api-reference/next.config.js/rewrites
```

See #15855 by @jaysoo

Depending on the complexity of your current proxy config this might not be a simple migration for some projects. But more importantly, it means that I do not have _a single proxy config for all the apps_ in my workspace anymore. I have both Next.js apps and React apps in my workspace and both share the same backends. Now I have to maintain two configurations.

## Motivation
see above, IMO it defeats the purpose of using Nx to harmonize configuration across several apps.

## Suggested Implementation
Revert the deprecation and keep the `proxyConfig` option

## Alternate Implementations
Maybe it is possible to provide an adapter that maps the `proxyConfig` to Next.js' proprietary Rewrite configuration?

**Comment by luchsamapparat** [comment]
@AgentEnder now that Nx 16 is out and `proxyConfig` is still available... does that mean that there's a chance that the deprecation might be reverted?

We are currently evaluating how we might migrate to Next.js' `rewrite` configuration and realize that it is not an equivalent replacement to the `proxyConfig`:

https://github.com/chimurai/http-proxy-middleware#http-proxy-options
https://nextjs.org/docs/pages/api-reference/next-config-js/rewrites#rewriting-to-an-external-url

For example, I don't see a way how I can rewrite request headers with Next.js.

**Comment by bojanbass** [comment]
I have to say I'm quite surprised how many bug happen with NX upgrades in the last half year and the deprecation of `proxyConfig` is surely a breaking one. In fact, there is still a warning but the logic is already removed, favoring Next CLI:
https://github.com/nrwl/nx/commit/3d76d95b29985656fcb01e25af1b27e0453e60c1#diff-6abc0d30cb71c3db586bfa9cb55cac2d6952705ce0d528b6e142a43d1e7dff88

As @luchsamapparat mentioned, this is a step back for NX, because is was always directed in having similar generic setups for different frameworks, and defining `proxyConfig` for `serve` executor is a generic way to do it, regardless if you use Angular, Vue, React etc.

Now the biggest problem is there is no alternative to the problem. Next URL rewrites are not the same thing because if you only apply them when `NODE_ENV=production`, you can't run it locally in production mode and have a backend proxy configured, and besides that there are problem with loading images because of trailing slash collision.

To get the right solution I was thinking about using `customServer` option using `@nx/next:custom-server`. First, the generator is broken with the message, because one of the latest migrations remove `root` option from `build` executor:
`Project xxx has invalid config. Did you generate it with "nx g @nx/next:app"?`
Then I reverted this `root` option and tried to run the app, but got another error:
```
Error: Cannot find module '.../dist/apps/xxx/main.js' imported from .../node_modules/@nx/js/src/executors/node/node-with-require-overrides.js
```

So I finally came to my single, hacky solution, this is to copy paste next `server` executor before it was introduced (the link above). Hopefully NX team finds a proper solution to this.

**Comment by ndcunningham** [comment]
We made the decision to remove the `proxyConfig` feature in order to enhance our interaction with Next.js CLI and leverage its API. Our previous approach of not directly using the Next CLI led to numerous issues, and we believe that sacrificing `proxyConf` support is a necessary trade-off for achieving greater overall stability.

Furthermore, it's important to note that the proxy feature was primarily applied only to the local development server.

**Comment by bojanbass** [comment]
I understand your choice, but if you are deprecating it, it's not enough to just point to the rewrite option, which is not the same and much less powerful. As mentioned, I managed to clone the old version of next serve executor and also with-nx plugin to be able to continue with development.

@ndcunningham, have you checked all the problems I listed above? And there's one even bigger problem, even if I manage to setup rewrites, there's a huge slowdown in bootstrapping and incremental reloading of the app using the new Next CLI, I would say it is at least two times slower than before. 

Wrapping this up, I would only ask if you have any working example of the alternative proxy setup in the new version? Should we go the other way around and expose next api endpoints as pages and forward the requests to the external api? 

Thank you for all you efforts!
